package.json

```diff
@@ -16,6 +16,7 @@
   "dependencies": {
     "dayjs": "^1.10.4",
     "http-status-codes": "^2.1.4",
+    "joi": "^17.4.0",
```

Comment:
nitpick: `joi` was built primarily for the `hapi` ecosystem and is mainly used for server-side validation.  it looks like you're using this for form validation on the client, so I'd recommend something like https://www.npmjs.com/package/yup.  if it's too much to change though, this is still probably fine.

Reply:
That's fine - I picked joi because I knew that you and Alex had exposure to it, but I've used yup at previous clients. I will make the switch.

Reply:
It was news to me, but seems that Joi is separated completely from the Hapi stack now, and it does ship with a [browser bundle](https://github.com/sideway/joi/blob/4d328216001d8c9c18cc280e133c95d24e876e45/package.json#L8). 

I'm fine either way with Joi, Yup, or Formik + Yup.

Reply:
I will switch over to using yup with [react-hook-form](https://react-hook-form.com/api) in a followup. I've had a lot of success with that hook library, and it can do some fancier validation that I am doing in my custom hook right now and we can easily switch between yup and joi (or other [validations libraries](https://react-hook-form.com/get-started#SchemaValidation)) if there is ever an issue.